firebase-perf/src/main/java/com/google/firebase/perf/application/AppStateMonitor.java

```diff
@@ -254,16 +256,43 @@ public void registerForAppState(WeakReference<AppStateCallback> client) {
    */
   /** @hide */
   public void unregisterForAppState(WeakReference<AppStateCallback> client) {
-    synchronized (mClients) {
-      mClients.remove(client);
+    synchronized (appStateClients) {
+      appStateClients.remove(client);
+    }
+  }
+
+  /**
+   * Register a client to receive app cold start update.
+   *
+   * @param client an AppColdStartCallback instance.
+   * @hide
+   */
+  /** @hide */
+  public void registerForAppColdStart(AppColdStartCallback client) {
+    synchronized (appStateClients) {
+      appColdStartClients.add(client);
```

Comment:
Should we check if the subscriber is already registered before adding to the list?

Reply:
Since we are adding to a `Set` it won't be problem.